Autocomplete/predefined-items-and-recent-searches/src/index.js

```diff
@@ -0,0 +1,64 @@
+import algoliasearch from "algoliasearch/lite";
+import {
+  autocomplete,
+  getAlgoliaHits,
+  reverseHighlightItem,
+} from "@algolia/autocomplete-js";
+import { predefinedItems } from "./predefined-items";
+import { recentSearches } from "./recent-searches";
+
+const searchClient = algoliasearch(
+  "latency",
+  "6be0576ff61c053d5f9a3225e2a90f76"
+);
+
+autocomplete({
+  container: "#autocomplete",
+  plugins: [predefinedItems, recentSearches],
+  openOnFocus: true,
+  debug: true,
+  onStateChange({ state, prevState }) {
+    if (state.query !== prevState.query) {
+      console.log("query:", state.query);
+    }
+  },
+  getSources({ query }) {
+    return [
+      {
+        getInputValue: ({ suggestion }) => suggestion.query,
+        getSuggestions({ query }) {
+          return getAlgoliaHits({
+            searchClient,
+            queries: [
+              {
+                indexName: "instant_search_demo_query_suggestions",
+                query,
+                params: {
+                  hitsPerPage: 4,
+                  facetFilters: [...recentSearches.data.getFacetFilters()],
+                },
+              },
+            ],
+          }).then((results) => {
```

Comment:
We can get rid of this once https://github.com/algolia/autocomplete.js/pull/331 is released.